sklearn/preprocessing/_target_encoder.py

```diff
@@ -321,26 +380,142 @@ def _fit_encodings_all(self, X, y):
             dtype=np.int64,
             count=len(self.categories_),
         )
+        if self.target_type_ == "multiclass":
+            encodings = self._learn_multiclass_encodings(
+                X_ordinal,
+                y,
+                n_categories,
+                self.target_mean_,
+            )
+            self.encodings_ = encodings
+        else:
+            self._learn_encodings(
+                X_ordinal,
+                y,
+                n_categories,
+                self.target_mean_,
+                save_encodings=True,
+            )
+
+        return X_ordinal, X_known_mask, y, n_categories
+
+    def _learn_encodings(self, X_ordinal, y, n_categories, target_mean, save_encodings):
```

Comment:
what about `_fit_encodings`..?

Reply:
I do not think `_fit_encodings` work because it is too generic. `_learn_encodings` learn encodings for a continuous and binary targets. `_learn_multiclass_encodings` calls `_learn_encodings` a few times with binary targets. 

Maybe `_learn_binary_or_continuous_encoding`? It's more verbose but the naming is consistent with `type_of_target`.

Reply:
Ah I follow better now. I can't really think of anything better than provides the context: "_learn_multiclass_encodings calls `_learn_encodings` a few times with binary targets."

Not really great but `_learn_single_target_encoding` and `_learn_multi_target_encoding` - because `_learn_multi_target_encoding` could really be made to work with multiple continuous targets. But since yours matches with `type_of_target` I think it is better.

Will amend to your suggestion and see if something better will be suggested by someone else.